test/e2e/apps/deployment.go

```diff
@@ -880,12 +884,15 @@ func testProportionalScalingDeployment(f *framework.Framework) {
 	err = framework.WaitForReplicaSetDesiredReplicas(c.ExtensionsV1beta1(), secondRS)
 	Expect(err).NotTo(HaveOccurred())
 
-	framework.Logf("Verifying that first rollout's replicaset has 11 available replicas")
-	err = framework.WaitForReplicaSetAvailableReplicas(c.ExtensionsV1beta1(), firstRS, 11)
+	biggerProportion := (newReplicas - replicas + 1) / 2
+	availableReplicas = availableReplicas + biggerProportion
+	framework.Logf("Verifying that first rollout's replicaset has %d available replicas", availableReplicas)
+	err = framework.WaitForReplicaSetAvailableReplicas(c.ExtensionsV1beta1(), firstRS, availableReplicas)
 	Expect(err).NotTo(HaveOccurred())
 
-	framework.Logf("Verifying that second rollout's replicaset does not have any available replicas")
-	Expect(secondRS.Status.AvailableReplicas).Should(Equal(int32(0)))
+	availableReplicas = (newReplicas-replicas)*2 - biggerProportion
```

Comment:
To clarify the formula, `newReplicas-replicas` is difference between old and new number of replicas after a scaling event. To cater for a scaling _down_ event, we should use `abs(newReplicas-replicas)`, but I did not do so in example here because the test does specify it is a scaling _up_ event.

Note that smaller proportion can be obtained from `newReplicas - replicas - biggerProportion`. As we have `(newReplicas - replicas) + (newReplicas - replicas - biggerProportion)`, it can be simplified as `(newReplicas-replicas)*2 - biggerProportion`.